### PR TITLE
Replaced once_cell with std::sync 

### PR DESCRIPTION
**Pull request by harsh-ps-2003**
Removing the use of once_cell crate as we have LazyLock from rust 1.80 release

### CODE REVIEW

**Comment by harsh-ps-2003** [comment]
Can you take a look @DaniPopes

File: crates/node/core/src/node_config.rs [hunk 1]
```diff
@@ -8,6 +8,9 @@ use crate::{
     dirs::{ChainPath, DataDirPath},
     utils::get_single_header,
 };
+
```
Comment (DaniPopes) [inline crates/node/core/src/node_config.rs:11]:
```suggestion
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [workspace.package]
 version = "1.0.3"
 edition = "2021"
-rust-version = "1.79"
+rust-version = "1.80.0"
```
Comment (DaniPopes) [inline Cargo.toml:4]:
```suggestion
rust-version = "1.80"
```

Comment (DaniPopes) [inline Cargo.toml:4]: Needs also update in README.md, please also look for all other `1.79` in the repo and update them

File: crates/node/core/src/node_config.rs [hunk 3]
```diff
@@ -22,6 +25,11 @@ use reth_storage_errors::provider::ProviderResult;
 use std::{net::SocketAddr, path::PathBuf, sync::Arc};
 use tracing::*;
 
+/// The default prometheus recorder handle. We use a global static to ensure that it is only
+/// installed once.
+pub static PROMETHEUS_RECORDER_HANDLE: LazyLock<PrometheusHandle> =
+    LazyLock::new(|| prometheus_exporter::install_recorder().unwrap());
```
Comment (DaniPopes) [inline crates/node/core/src/node_config.rs:31]: Not sure why this is here, it should be in `crates/node/metrics/src/recorder.rs`

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Comment by harsh-ps-2003** [comment]
Sorry! will have to close this PR to open a new one! Some weird git thing happened!

**Pull request closed without merging**